### PR TITLE
rgw: filter prefix should be used when generating expiration header

### PR DESCRIPTION
**Pull request by zhangsw**
Signed-off-by: zhang Shaowen <zhangshaowen_yewu@cmss.chinamobile.com>

Tracker: https://tracker.ceph.com/issues/45144

### CODE REVIEW

**Comment by zhangsw** [comment]
@cbodley pls help review this，thanks~

File: src/rgw/rgw_lc.cc [hunk 1]
```diff
@@ -1580,8 +1580,8 @@ std::string s3_expiration_header(
   for (const auto& ri : rule_map) {
     const auto& rule = ri.second;
     auto& id = rule.get_id();
-    auto& prefix = rule.get_prefix();
     auto& filter = rule.get_filter();
+    auto& prefix = filter.has_prefix()?filter.get_prefix():rule.get_prefix();
```
Comment (theanalyst) [inline src/rgw/rgw_lc.cc]: nit: spacing

**Comment by cbodley** [comment]
@zhangsw can you please add a s3test case for this?

**Comment by zhangsw** [comment]
Test case: https://github.com/ceph/s3-tests/pull/334

**Comment by zhangsw** [comment]
@mattbenjamin I've updated the commit message.

**Comment by mattbenjamin** [comment]
HI @zhangsw,
1) I tested this PR by hand, and I find it works great!
2) I tested the s3-tests case in https://github.com/ceph/s3-tests/pull/334 , and it doesn't quite work, because the test assertion uses dict.has_key(), which isn't in python3;  suggested fix in the PR
regards,
Matt

**Comment by ivancich** [comment]
Teuthology results -- no regressions noted:
* http://pulpito.ceph.com/ivancich-2020-09-02_13:40:42-rgw-wip-eric-testing-2-distro-basic-smithi/
* http://pulpito.ceph.com/ivancich-2020-09-02_13:41:02-rgw-wip-eric-testing-2-distro-basic-smithi/

**Comment by ivancich** [comment]
@zhangsw @theanalyst @mathslinux No backports are listed in the tracker. If backports appropriate please update the tracker.

**Pull request merged**